fgpyo/util/inspect.py

```diff
@@ -226,7 +228,7 @@ def dict_parse(dict_string: str) -> Dict[Any, Any]:
                 return types.make_enum_parser(type_)
             elif types.is_constructible_from_str(type_):
                 return functools.partial(type_)
-            elif isinstance(type_, type(type(None))):
+            elif type_ is type(None):  # noqa: E721
```

Comment:
can you please add a test?

Reply:
@jdidion following up on your question from Slack, I think this is the right approach.

as far as I'm aware, flake8 guards against `type(x) == int` and the like. We're doing equality checks against types above (e.g. `if type_ == bool`) and flake8 doesn't complain, I'm guessing because we don't call `type()` in the same line. 

If you wanted to avoid the `noqa`, you could declare `NoneType = type(None)` above (or in `fgpyo.types` and import it), then replace this line with `elif type_ is NoneType`. But this feels like an edge case in flake8, so I think what you have is fine if you're fine with it